integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestRangeColumnDataLoad.scala

```diff
@@ -836,15 +836,52 @@ class TestRangeColumnDataLoad extends QueryTest with BeforeAndAfterEach with Bef
   test("set and unset table property: range_column") {
     sql(
       """
-        | CREATE TABLE carbon_range_column6(id INT, name STRING, city STRING, age INT)
-        | STORED AS carbondata
+        | CREATE TABLE carbon_range_column6(id INT, name STRING, city STRING, age INT, bin binary,
+        |  bool1 boolean, arr1 array<int>, struct1 struct<id1:string,name1:string>,
+        |  map1 map<string,string>, dec1 decimal(10,5)) STORED AS carbondata
         | TBLPROPERTIES('SORT_SCOPE'='LOCAL_SORT', 'SORT_COLUMNS'='name, city')
       """.stripMargin)
 
     sql("ALTER TABLE carbon_range_column6 SET TBLPROPERTIES('range_column'='city')")
     sql("ALTER TABLE carbon_range_column6 SET TBLPROPERTIES('range_column'='name')")
     sql("ALTER TABLE carbon_range_column6 UNSET TBLPROPERTIES('range_column')")
     sql("ALTER TABLE carbon_range_column6 SET TBLPROPERTIES('range_column'='name')")
+
+    var ex = intercept[RuntimeException] {
+      sql("ALTER TABLE carbon_range_column6 SET TBLPROPERTIES('range_column'='bin')")
```

Comment:
Can move this alter exception validation to method. method could be validateInvalidRangecolumn(rangeColumn: String, dataType: String)

Reply:
Done, with different method signature.

Reply:
Done